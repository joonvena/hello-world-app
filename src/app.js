import express from 'express';
import bodyParser from 'body-parser';
import version from './api/version';

const app = express();
app.use(bodyParser.json());

app.use('/api/version', version);

app.close = () => { }

export default app;