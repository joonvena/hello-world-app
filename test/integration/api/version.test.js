
import supertest from 'supertest';
import app from '../../../src/app';

const request = supertest(app);

afterAll(() => app.close());

describe('/api/version', () => {
  describe('GET', () => {
    it('should return ok', async () => {
      const { body } = await request.get('/api/version').expect(200)

      expect(body).toMatchSnapshot();
    });
  });
});
