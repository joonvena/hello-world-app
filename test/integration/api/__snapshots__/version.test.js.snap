// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/version GET should return ok 1`] = `
Object {
  "status": "ok",
}
`;
